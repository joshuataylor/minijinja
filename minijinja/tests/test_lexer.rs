#![cfg(feature = "unstable_machinery")]
use minijinja::machinery::tokenize;

use std::fmt::Write;

#[test]
fn test_lexer() {
    insta::glob!("lexer-inputs/*.txt", |path| {
        let contents = std::fs::read_to_string(path).unwrap();
        let tokens: Result<Vec<_>, _> = tokenize(&contents, false).collect();
        let tokens = tokens.unwrap().into_iter().map(|x| x.0).collect::<Vec<_>>();
        insta::with_settings!({
            description => contents.trim_end(),
            omit_expression => true
        }, {
            let mut stringified = String::new();
            for token in tokens {
                writeln!(stringified, "{:?}", token).unwrap();
            }
            insta::assert_snapshot!(&stringified);
        });
    });
}
