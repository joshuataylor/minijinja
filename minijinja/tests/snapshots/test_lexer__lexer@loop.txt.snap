---
source: minijinja/tests/test_lexer.rs
description: "<ul>\n{% for item in sequence %}\n    <li>{{ item }}</li>\n{% endfor %}\n</ul>"
input_file: minijinja/tests/lexer-inputs/loop.txt
---
[
    TEMPLATE_DATA("<ul>\n"),
    BLOCK_END(false),
    IDENT(for),
    IDENT(item),
    IDENT(in),
    IDENT(sequence),
    BLOCK_END(false),
    TEMPLATE_DATA("\n    <li>"),
    VARIABLE_START(false),
    IDENT(item),
    VARIABLE_END(false),
    TEMPLATE_DATA("</li>\n"),
    BLOCK_END(false),
    IDENT(endfor),
    BLOCK_END(false),
    TEMPLATE_DATA("\n</ul>"),
]
